### PR TITLE
Add new member dialog to admin

### PR DESCRIPTION
**Pull request by nicolaskempf57**
Closes https://github.com/etalab/data.gouv.fr/issues/1336

It also renames `MembershipRequest` to `AdminMembershipRequest` to align with the naming of the other admin components.

It moves the edit member logic to its own components.

It also slightly improve the multiselect style to better match Figma.

### CODE REVIEW

**Review by Jorek57 — DISMISSED** [review]

File: udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                @submit.prevent="submit"
+              >
+                <div class="fr-modal__content">
+                  <h1
+                    :id="modalTitleId"
+                    class="fr-modal__title fr-mb-2w"
+                  >
+                    {{ t("Add member to the organization") }}
+                  </h1>
+                  <InputGroup
```
Comment (maudetes) [inline udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue:32]: We're waiting for the search component here right?

File: udata_front/theme/gouvfr/assets/js/components/AdminEditMember/AdminEditMemberButton.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    :aria-controls="modalId"
+  >
+    {{ t("Edit") }}
+  </button>
+  <AdminEditMemberModal
+    :id="modalId"
+    :member="member"
+    :oid="oid"
+    :roles="roles"
+    @member-updated="$emit('memberUpdated')"
```
Comment (maudetes) [inline udata_front/theme/gouvfr/assets/js/components/AdminEditMember/AdminEditMemberButton.vue:14]: I have some warnings in my browser:
```
Extraneous non-emits event listeners (memberUpdated) were passed to component ...
```

File: udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    :model-value="role"
+                    @update:model-value="(newRole) => role = (newRole as MemberRole)"
+                    :options="roles"
+                  />
+                </div>
+                <div class="fr-modal__footer">
+                  <div class="fr-btns-group fr-btns-group--right fr-btns-group--sm fr-btns-group--inline-lg fr-btns-group--icon-left">
+                    <button
+                      class="fr-btn fr-btn--secondary fr-btn--secondary-grey-500"
+                      @click="closeModal"
```
Comment (maudetes) [inline udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue:49]: The form gets wrongly submitted if clicking on Cancel button if some value has been set in input
```suggestion
                      type="button"
                      @click="closeModal"
```

File: udata_front/theme/gouvfr/assets/less/components/buttons.less [hunk 4]
```diff
@@ -87,7 +87,7 @@ These are now removed, see tags instead.
     border-radius: 3em;
 }
 
-.fr-btn--sm {
+.fr-btns-group--sm .fr-btn:not([class^=fr-icon-]):not([class*=" fr-icon-"]):not([class^=fr-fi-]):not([class*=" fr-fi-"]), .fr-btn--sm {
```
Comment (maudetes) [inline udata_front/theme/gouvfr/assets/less/components/buttons.less:90]: Oh wow, do we need to go with this complex selector?

**Review by maudetes — COMMENTED** [review]
Thank you for this PR :pray:

File: udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue (hunk 1 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue:32]: Yes ! The API call can be tested by providing an organization id in the input field in the meantime

File: udata_front/theme/gouvfr/assets/less/components/buttons.less (hunk 4 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/less/components/buttons.less:90]: This is the DSFR selector that we are overwritting so I'm using it as is. We can narrow it or use a custom logic but it won't be a perfect match anymore

File: udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue (hunk 3 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/AdminAddMember/AdminAddMemberModal.vue:49]: 🙏🏻

File: udata_front/theme/gouvfr/assets/less/components/buttons.less (hunk 4 above)
Comment (maudetes) [reply udata_front/theme/gouvfr/assets/less/components/buttons.less:90]: Nah let's keep it! It's just that I'm not completely fluent in CSS selector language ahah

Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/less/components/buttons.less:90]: The syntax is quite odd, I agree. It targets `.fr-btn` that have no icon related classes. `class^` matches a class attribute starting with `fr-icon-`  and `class*` matches a class attribute that contains ` fr-icon-`.

**Comment by nicolaskempf57** [comment]
Replaced by https://github.com/datagouv/udata-front/pull/442

**Pull request closed without merging**